### PR TITLE
metrics: rework metrics implementation

### LINKED ISSUE #3572: Support yaml as metrics file
Yaml is one of the most used file format. We need to support it for metrics.
More details: https://github.com/iterative/dvc/issues/3409#issuecomment-605444643

ToDo:
- [ ] update docs

### PR DESCRIPTION
**Pull request by efiop**
Dropping support for all types except json and yaml. No more xpath
support.

Fixes #3572

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. If the CLI API is changed, I have updated [tab completion scripts](https://github.com/iterative/dvc/tree/master/scripts/completion).

@dmpetrov agreed to provide docs PR for this. :pray: 

* [x] ❌ I will check DeepSource, CodeClimate, and other sanity checks below. (We consider them recommendatory and don't expect everything to be addressed. Please fix things that actually improve code or fix bugs.)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Comment by efiop** [comment]
Need to use custom loader for yaml for metrics and params to load float as Decimal. That way it will be much nicer looking and we won't have to bother with precision.

File: dvc/command/metrics.py [hunk 1]
```diff
@@ -224,19 +181,6 @@ def add_parser(subparsers, parent_parser):
...
-        help=(
-            "Type of metrics (json/yaml). "
-            "It can be detected by the file extension automatically. "
-            "Unsupported types will be treated as raw."
-        ),
-        metavar="<type>",
-    )
-    metrics_show_parser.add_argument(
-        "-x", "--xpath", help="json/yaml path.", metavar="<path>",
-    )
```
Comment (shcheklein) [inline dvc/command/metrics.py:239]: I thought we wanted to keep for metrics show (may be diff as well)?

**Comment by shcheklein** [comment]
Do we flatten the table in metrics show as we do for metrics diff now?

File: dvc/command/metrics.py (hunk 1 above)
Comment (efiop) [reply dvc/command/metrics.py:239]: Not in the same shape. As a filter - maybe, but later.

**Comment by efiop** [comment]
> Do we flatten the table in metrics show as we do for metrics diff now?

Not like in `metrics diff` but somewhat similar. `dvc metrics show` didn't get a lot of attention because of the upcoming `show` for experiments, so we agreed to just not touch it much for now.

**Pull request merged**